src/user/user.service.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException,
+  Post,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import * as bcrypt from 'bcrypt';

```

Comment:
use `bcrypt` in some helper service as this must be needed in multiple places, so that single function can does the purpose.